ide/api.lsp/src/org/netbeans/spi/lsp/StructureProvider.java

```diff
@@ -0,0 +1,244 @@
+         */
+        @NonNull
+        public Builder kind(@NonNull StructureElement.Kind name) {
+            this.kind = kind;
+            return this;
+        }
+
+        /**
+         * The detail of {@link StructureElement}, e.g the signature of a
+         * function. If not provided only the name is used.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder detail(@NonNull String detail) {
+            this.detail = detail;
+            return this;
+        }
+
+        /**
+         * The start of offset range where this element should be selected and
+         * revealed in document when selected in outline view.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder selectionStartOffset(int selectionStartOffset) {
+            this.selectionStartOffset = selectionStartOffset;
+            return this;
+        }
+
+        /**
+         * The end of offset range where this element should be selected and
+         * revealed in document when selected in outline view.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder selectionEndOffset(int selectionEndOffset) {
+            this.selectionEndOffset = selectionEndOffset;
+            return this;
+        }
+
+        /**
+         * The expanded range is offset range that is typically used to
+         * determine if the cursors inside the element to reveal in the element
+         * in the UI.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder expandedStartOffset(int expandedStartOffset) {
+            this.expandedStartOffset = expandedStartOffset;
+            return this;
+        }
+
+        /**
+         * The expanded range is offset range that is typically used to
+         * determine if the cursors inside the element to reveal in the element
+         * in the UI.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder expandedEndOffset(int expandedEndOffset) {
+            this.expandedEndOffset = expandedEndOffset;
+            return this;
+        }
+
+        /**
+         * Adds tag for this structure element.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder addTag(@NonNull StructureElement.Tag tag) {
+            if (this.tags == null) {
+                this.tags = new HashSet<>();
+            }
+            this.tags.add(tag);
+            return this;
+        }
+        
+        /**
+         * Tags for this structure element.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder tags(@NonNull Set<StructureElement.Tag> tags) {
+            this.tags = tags;
+            return this;
+        }
+
+        /**
+         * Adds sub element for this structure element.
+         *
+         * @since 1.8
+         */
+        @NonNull
+        public Builder addChild(@NonNull StructureElement child) {
```

Comment:
Why `addChild` vs. `children`? It'd be enough to have `children(StructureElement... children)` and possibly `children(List<StructureElement> children)` that would both add few child(ren) to the list.

Reply:
I have changed this.